force-app/main/default/classes/LeadTriggerTest.cls

```diff
@@ -0,0 +1,312 @@
+@IsTest
+public with sharing class LeadTriggerTest {
+
+    @TestSetup
+    static void setUpData() {
+
+        // create test account
+        Account testAccount = new Account(Name='Test Account');
+        insert testAccount;
+    
+        // create test contact
+        Contact testContact = new Contact(FirstName='Test', LastName='Contact', Email='test@example.com', AccountId=testAccount.Id);
+        insert testContact;
+    }
+
+    /*
+    * Test Title of a Lead has been update to Vice President, Manager, Assistant, Executive.
+    */
+    @IsTest
+    static void testHandleTitleNormalization_Insert() {
+
+        // creates leads        
+        List<Lead> ldList = new List<Lead>();
+
+        Lead leadVP = new Lead(FirstName = 'Test', LastName = 'Lead', Company = 'Test Company', Title = 'v.p.');
+        Lead leadMgr = new Lead(FirstName = 'Test', LastName = 'Lead', Company = 'Test Company', Title = 'manage');
+        Lead leadExec = new Lead(FirstName = 'Test', LastName = 'Lead', Company = 'Test Company', Title = 'chief');
+        Lead leadAssist = new Lead(FirstName = 'Test', LastName = 'Lead', Company = 'Test Company', Title = 'deputy');
+
+        ldList.add(leadVP);
+        ldList.add(leadMgr);
+        ldList.add(leadExec);
+        ldList.add(leadAssist);
+
+        // perform test
+        Test.startTest();
+        insert ldList;
+        Test.stopTest();
+
+        // get values to check things are working as they should
+        List<Lead> ldListCheckVP = [select Id, Title from Lead where Id = :leadVP.Id];
+        List<Lead> ldListCheckMgr = [select Id, Title from Lead where Id = :leadMgr.Id];
```

Comment:
Its better to use a single query to avoid prolonged runtime/limits. If you use a Map you can get the Lead record for each assert by Id. 

Something like:
```
Map<Id, Lead> leadMap = new Map<Id, Lead>([SELECT Id, Title FROM Lead]);

System.assertEquals('Manager', leadMap.get(leadMgr.Id).Title, 'Mgr Title is not Manager');
```